addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -829,8 +831,12 @@ var FieldMany2One = AbstractField.extend({
      * @override
      * @private
      */
-    _onKeydown: function () {
-        this.floating = false;
+    _onKeydown: function (ev) {
```

Comment:
I think you can remove this override. It has been added as a [fix](https://github.com/odoo/odoo/commit/8436150ae5820f1bde403c912ab1f9945e327d99) in 2017, but the many2one changed a lot since then, and it seems that this is no longer necessary (all m2o tests pass, and in the interface I couldn't reproduce the issue).